app/main.py

```
@@ -1,2 +1,145 @@
+
+        while self.hash_table[index] is not None:
+            if self.hash_table[index].key == key:
+                self.hash_table[index].value = values
+                return
+            index = (index + 1) % self.capacity
+
+        self.hash_table[index] = node
+        self.current_load += 1
+
+        if self.current_load >= self.threshold:
+            self.renew_capacity()
+
+    def __getitem__(self, key: Hashable) -> any:
+        index = hash(key) % self.capacity
+        while self.hash_table[index] is not None:
+            if self.hash_table[index].key == key:
+                return self.hash_table[index].value
+            index = (index + 1) % self.capacity
+        raise KeyError(f"KeyError: {key}")
+
+    def __len__(self) -> int:
+        return self.current_load
+
+    # _________________addition methods_____________________
+    def __iter__(self) -> any:
+        for index in range(self.capacity):
+            if self.hash_table[index] is not None:
+                yield self.hash_table[index].key
+
+    def clear(self) -> None:
+        self.capacity: int = 8
+        self.load_factor: float = 0.7
+        self.current_load: int = 0
+        self.threshold: float = self.capacity * self.load_factor
+        self.hash_table: list[Node | None | Iterable] = [None] * self.capacity
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = hash(key) % self.capacity
+        while self.hash_table[index] is not None:
+            if self.hash_table[index].key == key:
+                self.hash_table[index] = None
+                self.current_load -= 1
+                break
+        old_hash_table = [node for node in self.hash_table if node]
+        for node in old_hash_table:
+            index = node.hash_value % self.capacity
+            while self.hash_table[index] is not None:
+                index = (index + 1) % self.capacity
+            self.hash_table[index] = node
+        raise KeyError(f"KeyError: {key}")
```

The `__delitem__` method should raise a `KeyError` if the key is not found. Currently, it attempts to rehash without checking if the key was actually deleted, which can lead to incorrect behavior. Ensure to raise a `KeyError` if the key is not found before attempting to rehash.